Clustering/hybridEcalSplitter/src/DDStripSplitter.cc

```diff
@@ -0,0 +1,790 @@
+    _fout = new TFile("hybridCheckHistos.root","recreate");
+    h_phiModuleCheck = new TH2F("phi_HitModule","phi_HitModule", 20, -4, 16, 100, -2*TMath::Pi(), 2*TMath::Pi());
+
+    h_phiThetaMC = new TH2F("MCphiTheta", "MCphiTheta", 100, -2*TMath::Pi(), 2*TMath::Pi(), 100, -2*TMath::Pi(), 2*TMath::Pi());
+
+    h_stripDist_intercept   = new TH1F("dist_in","dist_in",100,0,2);
+    h_stripDist_nointercept = new TH1F("dist_no","dist_no",100,0,2);
+
+    for (int i=0; i<2; i++) {
+      TString reg = i==0 ? "_barrel" : "_endcap" ;
+      h_stavemodule[i] = new TH2F("stavemodule"+reg,"stavemodule"+reg,35, 0, 35, 35,0,35);;
+      h_layer[i]  = new TH1F("layer"+reg, "layer"+reg, 35,0,35);;
+
+      h_staveX[i]  = new TH2F("staveX_"+reg,  "staveX_"+reg,  100,-2000,2000,10,0,10);
+      h_staveY[i]  = new TH2F("staveY_"+reg,  "staveY_"+reg,  100,-2000,2000,10,0,10);
+      h_staveZ[i]  = new TH2F("staveZ_"+reg,  "staveZ_"+reg,  100,-5000,5000,10,0,10);
+      h_moduleX[i] = new TH2F("moduleX_"+reg, "moduleX_"+reg, 100,-2000,2000,10,0,10);
+      h_moduleY[i] = new TH2F("moduleY_"+reg, "moduleY_"+reg, 100,-2000,2000,10,0,10);
+      h_moduleZ[i] = new TH2F("moduleZ_"+reg, "moduleZ_"+reg, 100,-5000,5000,10,0,10);
+    }
+    
+    for (int ieb=0; ieb<2; ieb++) {
+      for (int is=0; is<10; is++) {
+	for (int im=0; im<10; im++) {
+	  TString hhname = ieb==0 ? "barrel" : "endcap";
+	  hhname+="_stave"; hhname+=is;
+	  hhname+="_module"; hhname+=im;
+	  h_cth_phi[ieb][is][im] = new TH2F(hhname+"_angle",hhname+"_angle",100,-1,1,100,-TMath::Pi(), TMath::Pi());
+	  h_XY[ieb][is][im] = new TH2F(hhname+"_xy",hhname+"_xy",100,-3000,3000,100,-3000,3000);
+	}
+      }
+    }
+  }
+
+  return;
+}
+
+void DDStripSplitter::processRunHeader( LCRunHeader*  /*run*/) { 
+   return;
+}
+
+
+
+void DDStripSplitter::setupGeometry() {
+
+
+  unsigned int includeFlag(0);
+  unsigned int excludeFlag(0);
+
+  includeFlag = ( dd4hep::DetType::CALORIMETER | dd4hep::DetType::ELECTROMAGNETIC | dd4hep::DetType::BARREL);
+  excludeFlag = ( dd4hep::DetType::AUXILIARY | dd4hep::DetType::FORWARD );
+
+  dd4hep::Detector & lcdd = dd4hep::Detector::getInstance();
+  const std::vector< dd4hep::DetElement>& theDetectors = dd4hep::DetectorSelector(lcdd).detectors(  includeFlag, excludeFlag ) ;
+  streamlog_out( DEBUG2 ) << " getExtension :  includeFlag: " << dd4hep::DetType( includeFlag ) << " excludeFlag: " << dd4hep::DetType( excludeFlag )
+			  << "  found : " << theDetectors.size() << "  - first det: " << theDetectors.at(0).name() << std::endl ;
+
+  if( theDetectors.size()  != 1 ){
+    std::stringstream es ;
+    streamlog_out (ERROR) << " getExtension: selection is not unique (or empty)  includeFlag: " << dd4hep::DetType( includeFlag ) << " excludeFlag: " << dd4hep::DetType( excludeFlag )
+			  << " --- found detectors : " ;
+    for( unsigned i=0, N= theDetectors.size(); i<N ; ++i ){
+      streamlog_out (ERROR) << theDetectors.at(i).name() << ", " ;
+    }
+    assert(0);
```

Comment:
Can you replace all `assert()` calls by `throw marlin::StopProcessingException(this)` ?

Reply:
done